#! /bin/sh /usr/share/dpatch/dpatch-run
## fix-open-or-die.dpatch by Diego Iastrubni <diego.iastrubni@xorcom.com>
##
## DP: Fix some perl syntax issues. Commited upstream at r3963, to be removed for 2.2.2 or 2.3.0

@DPATCH@
diff -urNad freepbx-2.2.0~dfsg~/amp_conf/bin/retrieve_iax_conf_from_mysql.pl freepbx-2.2.0~dfsg/amp_conf/bin/retrieve_iax_conf_from_mysql.pl
--- freepbx-2.2.0~dfsg~/amp_conf/bin/retrieve_iax_conf_from_mysql.pl	2007-02-26 11:23:37.000000000 +0000
+++ freepbx-2.2.0~dfsg/amp_conf/bin/retrieve_iax_conf_from_mysql.pl	2007-02-26 11:26:40.000000000 +0000
@@ -75,8 +75,8 @@
   exit;
 }
 
-open( EXTEN, ">$iax_conf" ) or die "Cannot create/overwrite IAX file: $iax_conf (!$)\n";
-open( REG, ">$iax_reg" ) or die "Cannot create/overwrite IAX Registration file: $iax_reg (!$)\n";
+open( EXTEN, ">$iax_conf" ) or die "Cannot create/overwrite IAX file: $iax_conf ($!)\n";
+open( REG, ">$iax_reg" ) or die "Cannot create/overwrite IAX Registration file: $iax_reg ($!)\n";
 
 print EXTEN $warning_banner;
 print REG   $warning_banner;
diff -urNad freepbx-2.2.0~dfsg~/amp_conf/bin/retrieve_op_conf_from_mysql.pl freepbx-2.2.0~dfsg/amp_conf/bin/retrieve_op_conf_from_mysql.pl
--- freepbx-2.2.0~dfsg~/amp_conf/bin/retrieve_op_conf_from_mysql.pl	2007-02-26 11:23:37.000000000 +0000
+++ freepbx-2.2.0~dfsg/amp_conf/bin/retrieve_op_conf_from_mysql.pl	2007-02-26 11:27:04.000000000 +0000
@@ -53,7 +53,7 @@
 
 	$zaplabel="Zap \%c";
 	$lastlabelnum=0;
-	open ZAPATA, "<$conffile" || die "Cannot open config file: $zapataconf\n";
+	open( ZAPATA, "<$conffile" ) or die "Cannot open config file: $zapataconf ($!)\n";
 	while( $line = <ZAPATA> ) {
 		next if $line =~ /^(\s)*$/;
 		chomp($line);
@@ -179,7 +179,7 @@
 	$dbh = DBI->connect("dbi:SQLite2:dbname=$db_file","","");
 }
 
-open( EXTEN, ">$op_conf" ) or die "Cannot create/overwrite config file: $op_conf (!$)\n";
+open( EXTEN, ">$op_conf" ) or die "Cannot create/overwrite config file: $op_conf ($!)\n";
 print EXTEN $warning_banner;
 
 #First, populate extensions
diff -urNad freepbx-2.2.0~dfsg~/amp_conf/bin/retrieve_parse_amportal_conf.pl freepbx-2.2.0~dfsg/amp_conf/bin/retrieve_parse_amportal_conf.pl
--- freepbx-2.2.0~dfsg~/amp_conf/bin/retrieve_parse_amportal_conf.pl	2007-02-26 11:23:40.000000000 +0000
+++ freepbx-2.2.0~dfsg/amp_conf/bin/retrieve_parse_amportal_conf.pl	2007-02-26 11:27:16.000000000 +0000
@@ -15,7 +15,7 @@
                 AMPENGINE => "asterisk",
 	);
 	
-	open(AMPCONF, $filename) or die "Cannot open $filename";
+	open(AMPCONF, $filename) or die "Cannot open $filename ($!)";
 	
 	while (<AMPCONF>)
 	{
diff -urNad freepbx-2.2.0~dfsg~/amp_conf/bin/retrieve_queues_conf_from_mysql.pl freepbx-2.2.0~dfsg/amp_conf/bin/retrieve_queues_conf_from_mysql.pl
--- freepbx-2.2.0~dfsg~/amp_conf/bin/retrieve_queues_conf_from_mysql.pl	2007-02-26 11:23:37.000000000 +0000
+++ freepbx-2.2.0~dfsg/amp_conf/bin/retrieve_queues_conf_from_mysql.pl	2007-02-26 11:27:31.000000000 +0000
@@ -67,7 +67,7 @@
   exit;
 }
 
-open( EXTEN, ">$queues_conf" ) or die "Cannot create/overwrite extensions file: $queues_conf (!$)\n";
+open( EXTEN, ">$queues_conf" ) or die "Cannot create/overwrite extensions file: $queues_conf ($!)\n";
 print EXTEN $warning_banner;
 
 $additional = "";
diff -urNad freepbx-2.2.0~dfsg~/amp_conf/bin/retrieve_sip_conf_from_mysql.pl freepbx-2.2.0~dfsg/amp_conf/bin/retrieve_sip_conf_from_mysql.pl
--- freepbx-2.2.0~dfsg~/amp_conf/bin/retrieve_sip_conf_from_mysql.pl	2007-02-26 11:23:37.000000000 +0000
+++ freepbx-2.2.0~dfsg/amp_conf/bin/retrieve_sip_conf_from_mysql.pl	2007-02-26 11:27:49.000000000 +0000
@@ -71,7 +71,7 @@
 	exit;
 }
 
-open( EXTEN, ">$sip_conf" ) or die "Cannot create/overwrite extensions file: $sip_conf (!$)\n";
+open( EXTEN, ">$sip_conf" ) or die "Cannot create/overwrite extensions file: $sip_conf ($!)\n";
 print EXTEN $warning_banner;
 
 $additional = "";
diff -urNad freepbx-2.2.0~dfsg~/amp_conf/bin/retrieve_zap_conf_from_mysql.pl freepbx-2.2.0~dfsg/amp_conf/bin/retrieve_zap_conf_from_mysql.pl
--- freepbx-2.2.0~dfsg~/amp_conf/bin/retrieve_zap_conf_from_mysql.pl	2007-02-26 11:23:37.000000000 +0000
+++ freepbx-2.2.0~dfsg/amp_conf/bin/retrieve_zap_conf_from_mysql.pl	2007-02-26 11:28:00.000000000 +0000
@@ -72,7 +72,7 @@
 	exit;
 }
 
-open( EXTEN, ">$zap_conf" ) or die "Cannot create/overwrite extensions file: $zap_conf (!$)\n";
+open( EXTEN, ">$zap_conf" ) or die "Cannot create/overwrite extensions file: $zap_conf ($!)\n";
 print EXTEN $warning_banner;
 
 $additional = "";
